valuable/src/structable.rs

```diff
@@ -7,15 +7,15 @@ pub trait Structable: Valuable {
     fn definition(&self) -> StructDef<'_>;
 }
 
-pub struct StructDef<'a> {
-    /// Type name
-    name: &'a str,
-
-    /// Fields
-    fields: Fields<'a>,
-
-    /// Is this a dynamic struct?
-    is_dynamic: bool,
+pub enum StructDef<'a> {
```

Comment:
should the enum itself also be `#[non_exhaustive]`? i can't imagine why we would add new variants to it, but it could be worth guarding against breakage?

Reply:
We could... it seems a bit extreme though, but I guess we could remove it later.